meta.go

```diff
@@ -832,6 +897,37 @@ func parseStorage(stores interface{}) map[string]Storage {
 	return result
 }
 
+func parseDevice(devices interface{}) (map[string]Device, error) {
+	if devices == nil {
+		return nil, nil
+	}
+	result := make(map[string]Device)
+	for name, device := range devices.(map[string]interface{}) {
+		deviceMap := device.(map[string]interface{})
+		deviceType, ok := deviceMap["type"].(string)
+		if !ok {
+			return nil, fmt.Errorf("%q has invalid device type", name)
```

Comment:
we should check that the value is valid also

Reply:
yes, u r right. we have the schema check and we have a `Check` func after `parse`. 
The reason I do this check here is I wanna ensure `DeviceType(deviceType)` will not panic if `deviceMap["type"]` is nil or non string